libavformat/avformat.h

```diff
@@ -1956,7 +1956,8 @@ typedef struct AVFormatContext {
      * SSIMWAVE
      * Callback to notify clients for ABR formats when an HTTP response has been received
      */
-    void (*http_response_code_callback)(void* opaque, const char* url, const char* method, int response_code);
+    void (*http_response_code_callback)(void* opaque, const int* av_stream_ids, size_t num_av_stream_ids,
```

Comment:
I chose to not pass AVStream pointers in the callback, in order to reduce external dependency on FFmpeg for the sake of unit testing.